contracts/src/subnet/SubnetActorCheckpointingFacet.sol

```diff
@@ -32,7 +32,13 @@ contract SubnetActorCheckpointingFacet is SubnetActorModifiers, ReentrancyGuard,
 
```

Comment:
We already have an `ensureValidCheckpoint` function that performs height validation based on the subnet actor state. We seem to be replicating some logic here, and in fact this logic is laxer than what we have there.

Please move all validation logic there and reconcile it.

Reply:
This has been done in the follow-up PR!